gradle-module-plugin/src/main/kotlin/io/ia/sdk/gradle/modl/task/SignModule.kt

```diff
@@ -332,40 +333,42 @@ open class SignModule @Inject constructor(_providers: ProviderFactory, _objects:
      */
     @Suppress("MemberVisibilityCanBePrivate")
     @Throws(IOException::class)
-    protected fun signModule(
-        keyStoreFile: File?,
-        pkcs11CfgFile: File?,
-        keystorePassword: String,
-        cert: File,
-        certPassword: String,
-        certAlias: String,
-        unsignedModule: File,
-        outFile: File,
-    ) {
+    protected fun signModule(unsignedModule: File) {
+        val keyStoreFile: File? = keystore.getOrNull()
+        val pkcs11CfgFile: File? = pkcs11Cfg.getOrNull()
+        val keystorePassword: String? = keystorePw.getOrNull()
+        val cert: File = certFile.get()
+        val certPassword: String? = certPw.getOrNull()
+        val certAlias: String = alias.get()
+        val outFile: File = signed.get().asFile
+
         logger.debug(
             "Signing module with keystoreFile: ${keyStoreFile?.absolutePath}, " +
                 "pkcs11CfgFile: ${pkcs11CfgFile?.absolutePath}, " +
                 "keystorePassword: ${"*".repeat(20)}, " +
                 "cert: ${cert.absolutePath}, " +
-                "certPw: ${"*".repeat(20)}, " +
+                "certPassword: ${"*".repeat(20)}, " +
                 "certAlias: $certAlias"
         )
 
         val keyStore: KeyStore = getKeyStore()
         loadKeyStore(keyStore, keystorePassword, keyStoreFile)
 
-        val privateKey: PrivateKey = keyStore.getKey(certAlias, certPassword.toCharArray()) as PrivateKey
+        val privateKey: PrivateKey = keyStore.getKey(
+            certAlias,
+            certPassword?.toCharArray()
+        ) as PrivateKey
 
         ModuleSigner(privateKey, cert.inputStream())
             .signModule(PrintStream(OutputStream.nullOutputStream()), unsignedModule, outFile)
     }
 
-    private fun loadKeyStore(ks: KeyStore, ksPwd: String, ksFile: File?) {
+    private fun loadKeyStore(ks: KeyStore, ksPwd: String?, ksFile: File?) {
         ksFile?.inputStream()
             // if PKCS#11 HSM (hardware key) keystore, forgo the input stream
             ?.takeUnless { ks.type == PKCS11_KS_TYPE }
             .use { maybeStream ->
-                ks.load(maybeStream, ksPwd.toCharArray())
+                ks.load(maybeStream, ksPwd?.toCharArray())
```

Comment:
`null` here is not a blocker even on PKCS#12 (only?) keystores with passwords; see the Javadoc for the impact.